src/Forms/TextCheckboxGroupField.php

```diff
@@ -32,6 +32,49 @@ public function __construct($title = null)
         $this->setTitle($title);
     }
 
+    public function getMessage()
+    {
+        $message = parent::getMessage();
+        if ($message) {
+            return $message;
+        }
+        foreach ($this->getChildren() as $field) {
+            $message = $field->getMessage();
+            if ($message) {
+                return $message;
+            }
+        }
```

Comment:
Should this concat the messages from multiple fields? At the moment it would skip the second message if there were two of them.

Reply:
I think it OK to skip. Templates are setup to only show a single validation message per field, so there would need to be further changes made to support multiple validation messages. In this case of this particular field, this field is a combination of the Title and ShowTitle fields, and since ShowTitle is a boolean checkbox field its impractical to have validation on it.

This code will make its way over to Composite field as part of [this card](https://github.com/silverstripe/silverstripe-framework/issues/11215) - it may make sense to look at multi validation messages there

Reply:
I agree with Michal here, sorry I didn't pick this up in my review, we could have saved some time.

> Templates are setup to only show a single validation message per field

While that is broadly true, this is an edge case where this isn't really *a field* in itself - it's really a holder for multiple fields. If each of the child fields has a validation error message, we should see those.

It would be bad UX to display only one validation message at a time, so users have to fix errors one field at a time, save, and see what fails next.

> since ShowTitle is a boolean checkbox field its impractical to have validation on it.

We can't really decide for developers what they should or shouldn't put validation on, unless we actually lock them out of doing so with exception messages when they try.

## Options
1. Have a new `getMessages()` method, and use that in the template for this form field, loop through the child messages and display all of them
2. Just update this method to concatenate the error messages together - at least the messages will display, even if it's not perfect
3. Explicitly defer this by adding ACs to the follow-up card you made. That's probably best, because there are additional things to consider such as the message type and cast, which may be different for the different messages.

Reply:
Go with 3, I've updated AC's on the follow up card